### ISSUE TITLE
ID Token undefined after renewing tokens

### ISSUE THREAD

**Issue by leonardo-asad** [AxaFrance/oidc-client#1270]
### Issue and Steps to Reproduce

After the initial authentication process I am able to get the id_token in my react application using the following hook:
```
const { idToken, idTokenPayload } = useOidcIdToken();
```

After 15 minutes, the access token expires and the tokens are renewed normally. **The body of the response does not include the ID Token**. After this if I try to get the idToken the value will be undefined and for the ID Token Payload the value will be null. 

I think that the problem might be that the library is expecting to have the idToken in the response body each time we renew the token. I researched and I found out that the idToken is optional in the response after successfully renewing the tokens:

![image](https://github.com/AxaFrance/oidc-client/assets/64209661/7c6d098b-2ec5-4574-92ec-b7ee46705d20)

### Versions

```
"@axa-fr/react-oidc": "^7.13.15"
```

### Screenshots
![image](https://github.com/AxaFrance/oidc-client/assets/64209661/b6dcae33-99dd-47f3-bef7-7effb270551c)

#### Expected
Being able to get the idToken and idToken Payload

#### Actual
After renewing the access token, the idToken is undefined and the idToken payload null.

Thank you !!

**Comment by guillaume-chervet** [comment]
Ho thank you for your issue @leonardo-asad .

It is the same for the refreshToken. I did not know that. I will try to fix it quickly !

**Comment by guillaume-chervet** [comment]
hi @leonardo-asad ,

You can test version 7.15.0-alpha.1286 that may fix that :)

**Comment by leonardo-asad** [comment]
hi @guillaume-chervet 

Thanks for your quick response !

It seems that this new version fixes the problem

Thanks!

**Comment by guillaume-chervet** [comment]
Version is published as [7.15.1](https://www.npmjs.com/package/@axa-fr/oidc-client/v/7.15.1) @leonardo-asad . Thank you very much for your help ! feel free to close the Issue :)

**Comment by OrtakProje-1** [comment]
Hi @guillaume-chervet 

I still encounter this problem.

When you go from the `parseOriginalTokens` method to the `setTokens` method, the condition `!('idToken' in tokens)` always returns `false`, because the `tokens` object always contains `idToken`.

![image](https://github.com/AxaFrance/oidc-client/assets/66565863/f11572e6-1944-41f2-af1f-e5eb66f0e7cb)

Please feel free to correct me if I'm wrong

**Versions**
`"@axa-fr/react-oidc": "^7.22.6"`
